src/org/zanata/jenkins/ScmGit.groovy

```diff
@@ -0,0 +1,33 @@
+package org.zanata.jenkins
+
+class ScmGit implements Serializable {
+  private def env
+  private def steps
+  private def mainRepoUrl
+
+  ScmGit(env, steps, mainRepoUrl) {
+    this.env = env
+    this.steps = steps
+    this.mainRepoUrl = mainRepoUrl
+  }
+
+  // getGitCommitId: get the commit Id of the tip of branch
+  //   branch: The branch of interested
+  //   repoUrl: Specify this if you are interested in other branch
+  //            (Such as pipeline-library)
+  public String getCommitId(String branch, String repoUrl = mainRepoUrl) {
+    String refString = null
+    if ( branch ==~ /PR-.*/ ) {
+      // Pull request does not show real branch name
+      refString = "refs/pull/" + env.CHANGE_ID + "/head"
+    } else {
+      refString = "refs/heads/" + branch
+    }
+    String commitLine = steps.sh([
+      returnStdout: true,
+      script: "git ls-remote " + repoUrl + " " + refString,
```

Comment:
Could you please add a comment here, something like this?

    // TODO race condition if branch changes; use checkout.GIT_COMMIT
    // See https://zanata.atlassian.net/browse/ZNTA-2237